src/Models/Model.php

```diff
@@ -354,6 +387,49 @@ public function jsonSerialize() {
 		return get_object_vars( $this );
 	}
 
+	/**
+	 * Constructs a model instance from database query data.
+	 *
+	 * @param object|array $queryData
+	 * @param int $mode The level of strictness to take when constructing the object, by default it will ignore extra keys but error on missing keys.
+	 * @return static
+	 */
+	public static function fromQueryData($queryData, $mode = self::BUILD_MODE_IGNORE_EXTRA): static {
```

Comment:
@borkweb This is the mode I was talking about. I'm curious what you think the default mode should be.